pyproject.toml

```diff
@@ -72,6 +77,14 @@ show_error_context = true
 disallow_untyped_calls = false
 exclude = ["noxfile.py", "dbt_core_interface.py"]
 
+[tool.poetry.extras]
+postgres = ["dbt-postgres", "psycopg2-binary"]
+
+[tool.poetry.plugins]
+
+[tool.poetry.plugins."sqlfluff"]
+"dbt_templater" = "dbt_core_interface.dbt_templater"
```

Comment:
No, there is no way in Python (AFAIK) for one package to disallow another one. You need to ensure that your environment setup does not install `sqlfluff-templater-dbt` in the first place if you want to use `dbt-core-interface` and the `/lint` endpoint.

Reply:
yeah, that's what I was thinking.  I thought the original issue was that we were using the same name.  is that still the case

Reply:
Using the same templater name (`dbt`) was a deliberate design decision. If we don't do this, then the user has to update their `.sqlfluff` configuration file to specify a different templater name in order to use the `/lint` endpoint.

To me, using the same name is the best way to resolve this tradeoff. What do you think?

Reply:
I'm open to ideas on this front, e.g. some sort of backdoor communication so the two templaters don't interfere with one another and the user gets what they expect.

My best idea on this front is to eliminate the custom templater by rewriting sqlfluff-templater-dbt to use this implementation, sharing the `DbtProjectContainer` singleton with the server if it's running inside a server, otherwise creating its own instance of the singleton.

I want to clarify, this PR is _literally recreating_ the same functionality we had in dbt-osmosis. I have not made any  changes to the design or implementation of the templater or the `/lint` endpoint, and I thought that was clear before I began the work.

I'm happy to have discussions along these lines, but I don't think this is the right context for those discussions. Redesigning the templater while porting it to a new environment risks introducing new issues. I recommend we focus on reviewing and testing whether this PR provides _the same functionality_ as dbt-osmosis. If we want to make changes, that can be follow-on work.